modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -262,13 +262,13 @@ public Object visit(PropertyIsLessThanOrEqualTo filter, Object extraData) {
     @Override
     public Object visit(PropertyIsLike filter, Object extraData) {
         BasicDBObject output = asDBObject(extraData);
-        String expr = convert(filter.accept(this, null), String.class);
+        String expr = convert(filter.getExpression().accept(this, null), String.class);
```

Comment:
If concat is not declared as supported in the filter capabiliities, it won't be part of the pre-filter. Unless there are other bugs, a FilterToSql subclass will only ever have to deal with bits that are declared in the store filter capabilities